include/wlr/types/wlr_keyboard.h

```diff
@@ -63,7 +63,7 @@ enum wlr_key_state {
 };
 
 struct wlr_event_keyboard_key {
-	uint32_t time_sec;
+	uint32_t time_msec;
```

Comment:
Does it have to be a uint32_t ?
2^32 msec is "only" 49-50 days, I could totally see folks (e.g. myself) keep a computer suspended/hibernated for that long, and wakeup might feel weird... Well I guess as far as wlroots is concerned it'll just be far over whatever threshold we're considering anyway, but if time_msec replaces both time_sec and time_usec, and is computed from time_usec, I'd keep it as uint64_t
(heck, I'd do everything as usec if it's not too much of a pain, but the design is around msecs so sure might as well go for it now)

Reply:
This comes from libinput (`libinput_event_keyboard_get_time_usec`), and I think the functions we pass msec to are our's (`cursor_update_position`, `wlr_seat_pointer_notify_axis`, `do_cursor_button_press`, ...)?
Have I missed timings from wayland itself?

I guess it's a big change just to refactor our code for that anyway so it's probably fine to leave as is, was just something to eventually keep in mind.

Reply:
>and I think the functions we pass msec to are our's

Yeah, but our functions don't really care about the time of these events, it's only used when forwarded to clients (`wlr_seat_pointer_notify_button`, `wlr_seat_pointer_notify_motion` and so on).

Reply:
Ok, as good as it will probably get as is then. Thanks for clearing it up.